### PR TITLE
Update class.md - add example of `new` returning a `PyResult`

### PR DESCRIPTION
**Pull request by aviramha**
It took me some time to realize how to return error in the new class API. I think it's better to have explicit documentation to save time for others :)

### CODE REVIEW

**Comment by messense** [comment]
See also https://pyo3.rs/v0.13.2/exception.html#handling-rust-errors

**Review by Eric-Arellano — COMMENTED** [review]
Note that this is already covered in the section right below: https://github.com/PyO3/pyo3/blob/main/guide/src/class.md#return-type

It's a bit easy to miss, but it has 3 hashtags compared to 2, meaning it's a subsection of the constructor section.

**Review by davidhewitt — APPROVED** [review]
I still think it's fine to explicitly call this out.

File: guide/src/class.md [hunk 1]
```diff
@@ -169,6 +169,23 @@ impl MyClass {
 }
 ```
 
+Alternatively, if your `new` method might throw errors you can return `PyResult<Self>`.
```
Comment (Eric-Arellano) [inline guide/src/class.md:172]: To be consistent with the below table, maybe word this as "may fail" instead of "might throw errors"?

Comment (aviramha) [reply guide/src/class.md]: Done

**Pull request merged**